ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1117,6 +1117,12 @@ public void testIndexSegmentRollOverKeySizeAndValueSizeChange() throws StoreExce
    */
   @Test
   public void indexPersistorTest() throws InterruptedException, IOException, StoreException {
+    // make sure persistor is persisting all segments
+    int numIndexSegments = state.index.getIndexSegments().size();
+    state.index.persistIndex();
+    int numIndexFiles = tempDir.listFiles(PersistentIndex.INDEX_SEGMENT_FILE_FILTER).length;
+    assertEquals("Not as many files written as there were index segments", numIndexSegments, numIndexFiles);
```

Comment:
Is there a way to use file access time or something to check for the order of persisting?

Reply:
I added a test but I'm not sure how useful it is. Though java says that the `File#lastModified()` provides a value in ms, it is limited by the underlying OS. On my mac, all I get are values to the closest second and the whole test runs in half a second, so it is not much use.

I put the test in a codebase that has the old code and it passes there too

The only way to conclusively test this would have been to insert hooks into the `IndexSegment` objects to determine when persist is called but with the current code setup where `PersistentIndex` creates these objects internally, there is no way to mock them or insert hooks.